src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/Editors/ImplicitUsingsEditor/CSharpImplicitUsingsControl.cs

```diff
@@ -0,0 +1,394 @@
+
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using Microsoft.VisualStudio.PlatformUI;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Controls
+{
+    internal sealed class CSharpImplicitUsingsControl : Control
+    {
+        public static readonly DependencyProperty AddUserUsingCommandProperty = DependencyProperty.Register(
+            nameof(AddUserUsingCommand),
+            typeof(ICommand),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata());
+
+        public static readonly DependencyProperty StringListProperty = DependencyProperty.Register(
+            nameof(StringList),
+            typeof(string),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(string.Empty, (o, e) => ((CSharpImplicitUsingsControl)o).OnStringListOrSupportedValuesPropertyChanged()));
+
+        public static readonly DependencyProperty UsingCollectionStateProperty = DependencyProperty.Register(
+            nameof(UsingCollectionState),
+            typeof(ObservableCollection<ImplicitUsingModel>),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(new ObservableCollection<ImplicitUsingModel>()));
+
+        public static readonly DependencyProperty EnteredUserUsingIncludesProperty = DependencyProperty.Register(
+            nameof(EnteredUserUsingIncludes),
+            typeof(string),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(string.Empty));
+
+        public static readonly DependencyProperty EnteredUserUsingAliasProperty = DependencyProperty.Register(
+            nameof(EnteredUserUsingAlias),
+            typeof(string),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(string.Empty));
+
+        public static readonly DependencyProperty EnteredUserUsingIsStaticProperty = DependencyProperty.Register(
+            nameof(EnteredUserUsingIsStatic),
+            typeof(bool),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(false));
+        
+        public static readonly DependencyProperty ShouldShowReadonlyUsingsProperty = DependencyProperty.Register(
+            nameof(ShouldShowReadonlyUsings),
+            typeof(bool),
+            typeof(CSharpImplicitUsingsControl),
+            new PropertyMetadata(true, (o, e) => ((CSharpImplicitUsingsControl)o).OnStringListOrSupportedValuesPropertyChanged()));
+        
+        // Used to suppress event handling during our own updates, breaking infinite loops.
+        private bool updating;
+
+        public CSharpImplicitUsingsControl()
+        {
+            UsingCollectionState = new ObservableCollection<ImplicitUsingModel>();
+
+#pragma warning disable VSTHRD012
```

Comment:
It would be good to have a comment explaining why it is OK to disable this warning here.